py/desisim/templates.py

```diff
@@ -222,14 +222,16 @@ def spectrum(self, oiiihbeta=None, oiihbeta=None, niihbeta=None,
         # Finally build the emission-line spectrum
         log10sigma = linesigma/LIGHT/np.log(10) # line-width [log-10 Angstrom]
         emspec = np.zeros(len(self.log10wave))
+
         for ii in range(len(line)):
             amp = line['flux'][ii] / line['wave'][ii] / np.log(10) # line-amplitude [erg/s/cm2/A]
             thislinewave = np.log10(line['wave'][ii] * (1.0+zshift))
             line['amp'][ii] = amp / (np.sqrt(2.0 * np.pi) * log10sigma)  # [erg/s/A]
 
             # Construct the spectrum [erg/s/cm2/A, rest]
-            emspec += amp * np.exp(-0.5 * (self.log10wave-thislinewave)**2 / log10sigma**2) \
-                      / (np.sqrt(2.0 * np.pi) * log10sigma)
+            jj = np.abs(self.log10wave-thislinewave) < 5*log10sigma
+            emspec[jj] += amp * np.exp(-0.5 * (self.log10wave[jj]-thislinewave)**2 / log10sigma**2) \
+                          / (np.sqrt(2.0 * np.pi) * log10sigma)
 
```

Comment:
I can't check right now if this would ever happen in the code, but you may want to make sure `jj` isn't sometimes false.  Also, can you make allowances for my idiosyncrasies and make it 6-sigma, please?

Reply:
I thought I was already being excessive in going to 5 sigma instead of 4 sigma, but ok, I changed it to 6.

Note that `jj` is an array of booleans, not a single boolean.  It is ok even if they are all `False`.  numpy will do the right thing and just not update any wavelengths in that case (i.e. not crash).  Does that address your concern, or am I missing something about your "make sure `jj` isn't sometimes false" suggestion?

```python
x = np.arange(10.0)
y = x**2

#- Some True, some False
ii = x<5
y[ii] += np.exp(-x[ii])

#- All False
jj  = x > 100
y[jj] += np.exp(-x[jj])
```

Reply:
Yes, of course.  All good, thanks.